keel-api/src/main/kotlin/com/netflix/spinnaker/keel/api/Constraint.kt

```diff
@@ -1,4 +1,6 @@
 package com.netflix.spinnaker.keel.api
 
-abstract class Constraint(val type: String)
+import com.netflix.spinnaker.keel.api.schema.Discriminator
+
+abstract class Constraint(@Discriminator val type: String)
```

Comment:
Any class that has sub-types that are determined by a discriminator property needs to have this annotation or the JSON schema generation will blow up.